apps/server/src/modules/files-storage/uc/file-record.uc.spec.ts

```diff
@@ -66,104 +95,180 @@ describe('FileRecordUC', () => {
 		expect(service).toBeDefined();
 	});
 
-	describe('fileRecordsOfParent', () => {
-		it('should call repo method findBySchoolIdAndTargetId with right parameters', async () => {
-			const { parentId } = fileParams;
-			const fileRecords = fileRecordFactory.buildList(3, { parentId, schoolId });
-			const spy = fileRecordRepo.findBySchoolIdAndParentId.mockResolvedValue([fileRecords, fileRecords.length]);
+	describe('getFileRecordsOfParent is called', () => {
+		describe('when user is authorised and valid files exist', () => {
+			const setup = () => {
+				const userId = new ObjectId().toHexString();
+				const { fileRecords, params } = getFileRecords();
 
-			await service.fileRecordsOfParent(userId, { schoolId, parentId, parentType: FileRecordParentType.School });
+				filesStorageService.getFilesOfParent.mockResolvedValueOnce([fileRecords, fileRecords.length]);
+				authorizationService.checkPermissionByReferences.mockResolvedValueOnce();
```

Comment:
Muss das explizit gemockt werden? Der Mock vom Service resolved das ohnehin schon oder?

Reply:
In einem der folgenden Tests auch noch mal

Reply:
Du meinst authorizationService? Da der mock nur einmalig gesetzt wird sollte ein Fehler fliegen wenn kein mock existiert. 

```
await this.checkPermission(userId, params.parentType, params.parentId, PermissionContexts.create);
const [fileRecords, count] = await this.filesStorageService.restoreFilesOfParent(params);
```

Reply:
Ja genau. Auch mit auskommentierter Zeile `authorizationService.checkPermissionByReferences.mockResolvedValueOnce();` läuft es noch erfolgreich durch. Aber das Szenario ist so vielleicht auch etwas klarer wenn man es explizit hinschreibt. Bleibt also erst mal so